src/Pagination/Pagination.tsx

```diff
@@ -43,21 +58,43 @@ export const Pagination = ({
     return null;
   }
 
+  // Total number of pages that can be tabbed through
   const totalPages = Math.ceil(totalDataLength / perPage);
+  // Total number of sets of pages (so how many sets of 5 can you go thru for example)
   const totalPageSets = Math.ceil(totalPages / maxPageButtonLimit);
+  // Which page set we're on
   const currentPageSet = Math.ceil(page / maxPageButtonLimit);
+  // lowest page of the page sets we are on
   const lowerBoundPage = (currentPageSet - 1) * maxPageButtonLimit + 1;
+  // highest page of the page set we are on
   const upperBoundPage = Math.min(
     currentPageSet * maxPageButtonLimit,
     totalPages,
   );
+
   if (page > totalPages || page < 1) {
     throw new Error(`Page must be between 1 and ${totalPages}`);
   }
 
   return (
     <Shadcn.Pagination className={cn('py-2', className)} {...paginationProps}>
       <Shadcn.PaginationContent>
+        {/* Items per page */}
+        {onPerPageChange && (
+          <>
+            <Shadcn.PaginationItem>
+              <span className="pr-1">Items per page:</span>
+            </Shadcn.PaginationItem>
+            <SelectInput
+              options={perPageOptions}
+              value={perPageOptions.filter((obj) => obj.value === perPage)}
```

Comment:
passing perPage is enough

Reply:
If you filter out the current perPage, we have no idea what perPage is currently selected.

Reply:
The reason I have it set that way is because if I only set it to `perPage`, the select doesn't show the default value and just shows "Select" even though 10 is selected since that's the default perPage value in the component.

Reply:
You can ignore the first comment since react-select v5 no longer supports `any` value/option types. 
I misunderstood your filter call, thought it was filtering out your options. Ignore these two comments.